### PR TITLE
New network plugin tests

### PR DESCRIPTION
**Pull request by GustavoYamaguchi**

### CODE REVIEW

File: src/test/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPluginTest.java [hunk 1]
```diff
@@ -76,46 +88,261 @@ public void validate() {
...
+		Mockito.doReturn("").when(this.httpRequestClientUtil)
+				.doPostRequest(Mockito.endsWith(OpenStackV2NetworkPlugin.SUFFIX_ENDPOINT_SECURITY_GROUP),
+						Mockito.eq(this.defaultToken), Mockito.any(JSONObject.class));
+		Mockito.doReturn(SECURITY_GROUP_ID).when(this.openStackV2NetworkPlugin).getSecurityGroupIdFromPostResponse(Mockito.anyString());
+		//post ssh rule
+		Mockito.doReturn("").when(this.httpRequestClientUtil)
+				.doPostRequest(Mockito.endsWith(OpenStackV2NetworkPlugin.SUFFIX_ENDPOINT_SECURITY_GROUP_RULES),
+						Mockito.eq(this.defaultToken), Mockito.any(JSONObject.class));
+		//post icmp rule
+		Mockito.doReturn("").when(this.httpRequestClientUtil)
```
Comment (arn3tt) [inline src/test/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPluginTest.java]: Do we need this duplication? Or is line 111 already specifying this?

File: src/main/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPlugin.java [hunk 2]
```diff
@@ -185,32 +185,33 @@ private void createSecurityGroupRules(NetworkOrder order, Token localToken, Stri
...
         }
     }
 
-    protected String getSecurityGroupIdFromJson(String json) throws UnexpectedException {
+    protected String getSecurityGroupIdFromPostResponse(String json) throws UnexpectedException {
         String securityGroupId = null;
         try {
             JSONObject rootServer = new JSONObject(json);
             JSONObject securityGroupJSONObject = rootServer.optJSONObject(KEY_SECURITY_GROUP);
             securityGroupId = securityGroupJSONObject.optString(KEY_ID);
```
[the remark this answers is not in the document]
Comment (GustavoYamaguchi) [reply src/main/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPlugin.java]: I guess that all of than are mandatory, so I replaced both with getJSONObject/getString can you review this part again?

File: src/main/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPlugin.java [hunk 3]
```diff
@@ -185,32 +185,33 @@ private void createSecurityGroupRules(NetworkOrder order, Token localToken, Stri
...
         return securityGroupId;
     }
 
-    private String getSecurityGroupIdFromResponseJson(String json) throws UnexpectedException {
+    protected String getSecurityGroupIdFromGetResponse(String json) throws UnexpectedException {
         String securityGroupId = null;
         try {
             JSONObject response = new JSONObject(json);
             JSONArray securityGroupJSONArray = response.optJSONArray(KEY_SECURITY_GROUPS);
             JSONObject securityGroup = securityGroupJSONArray.optJSONObject(0);
```
[the remark this answers is not in the document]
Comment (GustavoYamaguchi) [reply src/main/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPlugin.java]: I guess that all of than are mandatory, so I replaced both with getJSONObject/getString can you review this part again?

File: src/test/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPluginTest.java (hunk 1 above)
Comment (GustavoYamaguchi) [reply src/test/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPluginTest.java]: Each security group rule requires a new request. So, when we create a security group we need to add 2 new rules, allow SSH and ICMP, that's the reason for the 'duplicated' code, we're mocking both requests.

Comment (GustavoYamaguchi) [reply src/test/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPluginTest.java]: Fixed, thanks.

**Pull request merged**